pyard/misc.py

```diff
@@ -20,12 +21,43 @@ def get_n_field_allele(allele: str, n: int, preserve_expression=False) -> str:
 
 
 def get_3field_allele(a: str) -> str:
+    last_char = a[-1]
+    if last_char in PandG_chars:
+        a = a[:-1]
+    
     return get_n_field_allele(a, 3)
 
 
 def get_2field_allele(a: str) -> str:
+    last_char = a[-1]
+    if last_char in PandG_chars:
+        a = a[:-1]
     return get_n_field_allele(a, 2)
 
 
 def number_of_fields(allele: str) -> int:
     return len(allele.split(":"))
+
+# computes a valid G name based on the ambiguity string
+def get_G_name(a: str) -> str:
+    a = a.split('/')[0]
+    last_char = a[-1]
+    if last_char in PandG_chars + expression_chars:
+        a = a[:-1]
+    if len(a.split(':')) ==2:
```

Comment:
Can you add comments on what/why these 2 constructions are ?

Reply:
Which 2? PandG_chars and expression_chars?  They were already there, I'm just reusing them